sentry-android-ndk/src/main/java/io/sentry/android/ndk/DebugImagesLoader.java

```diff
@@ -75,7 +77,74 @@ public DebugImagesLoader(
     return debugImages;
   }
 
-  /** Clears the caching of debug images on sentry-native and here. */
+  /**
+   * Loads debug images for the given set of addresses.
+   *
+   * @param addresses Set of memory addresses to find debug images for
+   * @return Set of debug images, or null if debug images couldn't be loaded
+   */
+  public @Nullable Set<DebugImage> loadDebugImagesForAddresses(@NotNull Set<Long> addresses) {
+    try (final @NotNull ISentryLifecycleToken ignored = debugImagesLock.acquire()) {
+      List<DebugImage> allDebugImages = loadDebugImages();
+      if (allDebugImages == null) {
+        return null;
+      }
+
+      Set<DebugImage> referencedImages = new HashSet<>();
+      for (Long addr : addresses) {
+        DebugImage image = findImageByAddress(addr, allDebugImages);
+        if (image != null) {
+          referencedImages.add(image);
+        }
+      }
+
+      if (referencedImages.isEmpty()) {
+        options
+            .getLogger()
+            .log(
+                SentryLevel.WARNING,
+                "No debug images found for any of the %d addresses.",
+                addresses.size());
+        return null;
+      }
+
+      return referencedImages;
+    }
+  }
+
+  /**
+   * Finds a debug image containing the given address using binary search. Requires the images to be
+   * sorted.
+   *
+   * @return The matching debug image or null if not found
+   */
+  private @Nullable DebugImage findImageByAddress(long address, List<DebugImage> images) {
```

Comment:
Is there a specific reason you're assuming that the list of debug images is sorted? I [see we have quite a few different implementations](https://github.com/getsentry/sentry-native/tree/master/src/modulefinder), but I don't think there's any guarantee here.

Reply:
pretty much from testing manually it was always sorted on my end

but tbh the performance gains of using binary search are marginal at best since the number of images are not that big, I wouldn't mind switching the impl to just search for it normally to make it more stable if we're not confident it's always sorted

Reply:
Alright I just talked to Mischan, on Android/Linux based flavors the list should always be sorted. I still made some adaptions to account for a nullable `image.getImageSize()`